File: OBJCSample/OBJCSample/classes/AZCalendar.m
//
//  AZCalendar.m
//  AZCalendarOBJC
//
//  Created by Ali on 4/14/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

#import "AZCalendar.h"
#import "AZGregorianCalendar.h"
#import "AZPersianCalendar.h"
#import "AZHijriCalendar.h"
#import "AZMonthName.h"
@implementation AZCalendar

+ (DateType*) getCurrentGregorianDate{
    
    
    NSInteger miladiDay =  [AZGregorianCalendar getGregorianDayNumber];
    NSInteger miladiMonth =  [AZGregorianCalendar getGregorianMonthNumber];
    NSInteger miladiYear =  [AZGregorianCalendar getGregorianYearNumber];
    
    return [[DateType alloc] initWithDay:miladiDay month:miladiMonth year:miladiYear];
    
}
+ (DateType*) getCurrentPersianDate{
    NSInteger persianDay =  [AZPersianCalendar getPersianDayNumber];
    NSInteger persianMonth =  [AZPersianCalendar getPersianMonthNumber];
    NSInteger persianYear =  [AZPersianCalendar getPersianYearNumber];
    
    return [[DateType alloc] initWithDay:persianDay month:persianMonth year:persianYear];
}
+ (DateType*) getCurrentHijriDate{
    
    NSInteger hijriDay =  [AZHijriCalendar getHijriDayNumber];
    NSInteger hijriMonth =  [AZHijriCalendar getHijriMonthNumber];
    NSInteger hijriYear =  [AZHijriCalendar getHijriYearNumber];
    
    return [[DateType alloc] initWithDay:hijriDay month:hijriMonth year:hijriYear];
}

+(DateType*) convertDateWithInputType:(CalendarType)inputType inputDate:(DateType*)inputDate outputType:(CalendarType)outputType{

    return  [AZConvertorCalendar convertDateWithinputType:inputType inputDate:inputDate outputType:outputType];
}

+(NSInteger)getDayIndexWithDate : (DateType*)date calendar:(CalendarType)calendar{
    NSInteger result = 0;
    
    NSCalendarIdentifier type = NSCalendarIdentifierPersian;
    
    
    switch (calendar) {
        case persian :
            type = NSCalendarIdentifierPersian;
            break;
        case gregorian :
            type = NSCalendarIdentifierGregorian;
            break;
        case hijri :
            type = NSCalendarIdentifierIslamicCivil;
            break;
            
        default:
            break;
    }
    
    
    NSCalendar *myCal = [[NSCalendar alloc] initWithCalendarIdentifier:type];
    
    NSDateComponents *component = [[NSDateComponents alloc] init];
    component.day = date.day;
    component.month = date.month;
    component.year = date.year;
    
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    formatter.dateFormat = @"EEE";
    
    NSDate *myDate = [myCal dateFromComponents:component];
    
    
   

    if ([[formatter stringFromDate:myDate] isEqualToString:@"Sun"]){
        result = 0;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Mon"]){
        result = 1;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Tue"]){
        result = 2;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Wed"]){
        result = 3;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Thu"]){
        result = 4;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Fri"]){
        result = 5;
    }
    if ([[formatter stringFromDate:myDate] isEqualToString:@"Sat"]){
        result = 6;
    }
    
    
    
 
    
    
    return  result;
}

+(NSString*)getDayNameWithCalendar : (CalendarType)calendar date:(DateType*)date{
    NSString *result = @"Oops :|";
    NSCalendarIdentifier type = NSCalendarIdentifierPersian;
    switch (calendar) {
        case persian :
            type = NSCalendarIdentifierPersian;
            break;
        case gregorian :
            type = NSCalendarIdentifierGregorian;
            break;
        case hijri :
            type = NSCalendarIdentifierIslamicCivil;
            break;
            
        default:
            break;
    }

    
    NSCalendar *myCal = [[NSCalendar alloc] initWithCalendarIdentifier:type];
    
    NSDateComponents *component = [[NSDateComponents alloc] init];
    component.day = date.day;
    component.month = date.month;
    component.year = date.year;
    
    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
    
    formatter.dateFormat = @"EEE";
   
    NSDate *myDate = [myCal dateFromComponents:component];
    
    result = [formatter stringFromDate:myDate];
   
    return result;
}

+(NSString*)getMonthNameWithCalendar : (CalendarType)calendar date:(DateType*)date{
    
    return  [AZMonthName getMonthNameWithCalendar:calendar date:date] ;
}

@end

File: SwiftSample/SwiftSample/Framework/AZCalendarSwift.framework/Headers/AZCalendar.swift
//
//  AZCalendar.swift
//  AZCalendarSwift
//
//  Created by Ali on 4/11/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

import Foundation
public class AZCalendar {
    
    
   public static func getCurrentGregorianDate() -> DateType{
        
        let miladiDay = AZGregorianCalendar.getGregorianDayNumber()
        let miladiMonth = AZGregorianCalendar.getGregorianMonthNumber()
        let miladiYear = AZGregorianCalendar.getGregorianYearNumber()
        
        return DateType.init(day: miladiDay, month: miladiMonth, year: miladiYear)
        
    }
   public static func getCurrentPersianDate() -> DateType{
        
        let persianDay = AZPersianCalendar.getPersianDayNumber()
        let persianMonth = AZPersianCalendar.getPersianMonthNumber()
        let persianYear = AZPersianCalendar.getPersianYearNumber()
        
        return DateType.init(day: persianDay, month: persianMonth, year: persianYear)
    }
  public  static func getCurrentHijriDate() -> DateType {
    
    let hijriDay = AZHijriCalendar.getHijriDayNumber()
    let hijriMonth = AZHijriCalendar.getHijriMonthNumber()
    let hijriYear = AZHijriCalendar.getHijriYearNumber()
    
    return DateType.init(day: hijriDay, month: hijriMonth, year: hijriYear)
    
    }
    
    public static func  convertDate(inputType : CalendarType , inputDate : DateType , outputType : CalendarType)->DateType {
    
        let result = AZConvertorCalendar.convertDate(inputType: inputType, inputDate: inputDate, outputType: outputType)
        return result
    }
    
    public static func getDayIndex (date : DateType , calendar : CalendarType) -> Int {
    var result = 0

        var type = Calendar.Identifier.persian
        

        
        switch calendar {
        case .persian:
            type = .persian
            break
        case .gregorian :
            type = .gregorian
            break
        case .hijri :
            type = .islamicCivil
            break
        }
        
      
        
        let myCal = Calendar.init(identifier: type)
        
        
        var component = DateComponents.init()
        component.day = date.day
         component.month = date.month
         component.year = date.year
        
       let formatter = DateFormatter.init()
        formatter.dateFormat = "EEE"
        
        
       
        
        let myDate = myCal.date(from: component)
        
       
        switch formatter.string(from: myDate!) {
        case "Sun":
            result = 0
            break
        case "Mon":
            result = 1
            break
        case "Tue":
            result = 2
            break
        case "Wed":
            result = 3
            break
        case "Thu":
            result = 4
            break
        case "Fri":
            result = 5
            break
        case "Sat":
            result = 6
            break
        default:
            break
        }
        
        
        
    return result
    }
    
    public static func getDayName (calendar : CalendarType , date : DateType) -> String {
        var result = "Oops :|"
        var type = Calendar.Identifier.persian
        
        switch calendar {
        case .persian:
            type = .persian
            break
        case .gregorian :
            type = .gregorian
            break
        case .hijri :
            type = .islamicCivil
            break
        }
        
      
        
        let myCal = Calendar.init(identifier: type)
        
        
        var component = DateComponents.init()
        component.day = date.day
        component.month = date.month
        component.year = date.year
        
        let formatter = DateFormatter.init()
        formatter.dateFormat = "EEE"
            let myDate = myCal.date(from: component)
        result = formatter.string(from: myDate!)
        return result
    }
    
    public static func getMonthName (calendar : CalendarType , date : DateType) -> String{
        
       
       return AZMonthName.getMonthName(calendar: calendar, date: date)
        
        
       }
    
    
}
